admin-dev/themes/new-theme/js/pages/product/components/combination-modal/CombinationModal.vue

```diff
@@ -0,0 +1,282 @@
+          return;
+        }
+        this.selectedCombinationId = null;
+      },
+      showPrevious() {
+        if (this.previousCombinationId !== null) {
+          this.selectedCombinationId = this.previousCombinationId;
+        }
+      },
+      showNext() {
+        if (this.nextCombinationId !== null) {
+          this.selectedCombinationId = this.nextCombinationId;
+        }
+      },
+      submitForm() {
+        this.submittingCombinationForm = true;
+        const iframeBody = this.$refs.iframe.contentDocument.body;
+        const editionForm = iframeBody.querySelector(ProductMap.combinations.editionForm);
+        editionForm.submit();
+      },
+    },
+    watch: {
+      selectedCombinationId(combinationId) {
+        if (combinationId === null) {
+          this.previousCombinationId = null;
+          this.nextCombinationId = null;
+          this.editCombinationUrl = null;
+
+          return;
+        }
+
+        this.loadingCombinationForm = true;
+        this.editCombinationUrl = router.generate('admin_products_combinations_edit_combination', {
+          combinationId,
+          liteDisplaying: 1,
+        });
+
+        const selectedIndex = this.combinationIds.indexOf(combinationId);
+
+        if (selectedIndex === -1) {
+          this.previousCombinationId = null;
+          this.nextCombinationId = null;
+        } else {
+          this.previousCombinationId = selectedIndex === 0 ? null : this.combinationIds[selectedIndex - 1];
+          this.nextCombinationId = selectedIndex === this.combinationIds.length - 1
+            ? null
+            : this.combinationIds[selectedIndex + 1];
+        }
+      },
+    },
+  };
+</script>
+
+<style lang="scss" type="text/scss">
+@import "~@scss/config/_settings.scss";
+
+#combination-edit-modal {
+  .modal-dialog {
+    max-width: 1200px;
+    width: 90%;
+    height: 95%;
+    margin: 0 auto;
+
+    .modal-header {
+      display: none;
+    }
+
+    .modal-content {
+      height: 100%;
+      padding: 0;
+      margin: 0;
+
+      .modal-body {
+        padding: 0;
+        margin: 0;
+
+        .combination-loading {
+          position: absolute;
+          width: 100%;
+          height: 100%;
+          display: flex;
+          align-items: center;
+          justify-content: center;
+          z-index: 1;
+          background: rgba(255, 255, 255, 0.8);
+        }
+
+        .combination-iframe {
+          padding: 0;
+          margin: 0;
+          border: 0;
+          outline: none;
+          vertical-align: top;
+          width: 100%;
+          height: 100%;
+          display: block;
+
+          .card {
+            margin-bottom: 0;
+          }
+        }
+      }
+
+      .modal-footer {
+        margin: 0;
+        padding: 0.6rem 1rem;
+        display: flex;
+        flex-direction: row;
+        justify-content: flex-end;
+
+        .btn-close {
+          margin-right: auto;
+        }
+      }
+    }
+  }
+}
```

Comment:
it will no longer be reusable?

Reply:
Yes, if we use a vue component, include everything to the .vue is fine to me, btw what do you mean about reusable @PululuK ? By external users?

Reply:
I'm talking about css code.
Can't we put it in a dedicated css file?

Reply:
What's the usecase ?